tests/api/v1/generators/putWithCollector.js

```diff
@@ -189,4 +196,164 @@ describe('tests/api/v1/generators/putWithCollector.js >', () => {
       return done();
     });
   });
+
+  describe('assign to collector >', () => {
+    // generator already created with collector1 as possible collector
+    let clock;
+    const now = Date.now();
+    let collectorAlive1 = {
+      name: 'IamAliveAndRunning1',
+      version: '1.0.0',
+      status: 'Running',
+      lastHeartbeat: now,
+    };
+
+    let collectorAlive2 = {
+      name: 'IamAliveAndRunning2',
+      version: '1.0.0',
+      status: 'Running',
+      lastHeartbeat: now,
+    };
+
+    beforeEach(() => {
+      clock = sinon.useFakeTimers(now);
+    });
+
+    before((done) => { // create alive collectors
+      tu.db.Collector.bulkCreate([collectorAlive1, collectorAlive2])
+      .then((collectors) => {
+        collectorAlive1 = collectors[0];
+        collectorAlive2 = collectors[1];
+        return done();
+      })
+      .catch(done);
+    });
+
+    afterEach(() => clock.restore());
+
+    it('unassigned generator, put generator with possible collectors should ' +
+      'set currentCollector if alive collector is available', (done) => {
```

Comment:
Will this same case work on patch, since we're not mocking it there? Can you add a test for patch as well, just for this one case where a possible collector is added to an unassigned generator?